dynawo/sources/Modeler/DataInterface/DYNStateVariable.cpp

```diff
@@ -203,6 +205,18 @@ StateVariable::setValue(const double& value) {
     value_ = static_cast<int> (value);
     break;
   }
+  valueAffected_ = true;
+  rawValue_ = value;
+}
+
+void
+StateVariable::setBackupValue() {
+  backupValue_ = rawValue_;
+}
+
+bool
+StateVariable::hasValueChanged(const double fromValue) const {
```

Comment:
Wouldn't it be easier to remove the argument and compare to the backup value?

Reply:
No ! This function checks if value has changed **from a specific value**.